src/service/resources-impl.js

```diff
@@ -866,12 +869,23 @@ export class Resources {
         const resource = request.resource;
         const box = resource.getLayoutBox();
         const iniBox = resource.getInitialLayoutBox();
-        const diff = request.newHeight - box.height;
+        const margins = resource.getLayoutMargins();
+        const topMarginDiff = request.newMargins ?
+            request.newMargins.top - margins.top : 0;
+        const bottomMarginDiff = request.newMargins ?
+            request.newMargins.bottom - margins.bottom : 0;
+        const heightDiff = request.newHeight - box.height;
+        // The lowest boundary of the element that would appear to be resized
+        // as a result of this size change. If the bottom margin is being
+        // changed then it is the bottom edge of the margin box, otherwise it
+        // is the bottom edge of the layout box.
+        const bottomDisplacedBoundary = bottomMarginDiff ?
+            box.bottom + margins.bottom : box.bottom;
```

Comment:
Ok, got it. Let's do margin changes and I'll review this one more time - it should be good to go.